### DISCUSSION TITLE
When using Swiper React + loop, onSlideChange is called twice sometimes.

### DISCUSSION THREAD

**Discussion by real-felix** [nolimits4web/swiper#6260] [Q&A] [answered]
Is that a bug (I guess so)? Is it already reported?

Suppose that I have 3 slides A, B and C; and that I swipe always to the right. The callback `onSlideChange` is called as follow:

A → B → C → A → A/B

As you can see, the callback is called twice after successfully looping. Is there a workaround?

BTW, it's the same with `onActiveIndexChange` .

**Comment by real-felix** [comment, marked as answer]
I fixed it by using `onRealIndexChange`.

The API is really confusing sometimes. It looks like public API as mixed with implementation details, it's really hard to understand what is relevant and what is not.

**Reply by real-felix** [reply]
BTW, it doesn't run the first time the swiper is displayed, so I had to do the following: `    <Swiper loop={true} onInit={doSomeStuff} onRealIndexChange={doSomeStuff}>`

**Comment by say8425** [comment]
I agree that `onRealIndexChange` works more accurately than `onSlideChange`. 
However, if you want it to trigger after the user finishes a quick swipe, I recommend considering the use of `onTransitionEnd`.


```tsx
const [currentSlideIndex, setCurrentSlideIndex] = useState(initialSlideIndex);

return (
  <Swiper
    onTransitionEnd={({ activeIndex }) => {
      setCurrentSlideIndex(activeIndex);
    }}
  ></Swiper>
);
```
